docs-gen/content/docs/examples/pipeline_dependencies.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "Pipeline Dependencies"
+weight: 2
+---
+
+Pipeline dependencies allow splitting up larger machine learning pipelines into sub-pipelines. This is particularly useful when:
+- The data of an earlier step changes at a lower frequency than the data for subsequent steps
+- Outputs of an earlier step could be shared between pipelines to avoid re-processing the same data
```

Comment:
AFAIK another notable use case is to train two models and combine them into one for serving.